src/CodeGen_LLVM.cpp

```diff
@@ -2748,6 +2741,12 @@ void CodeGen_LLVM::visit(const Evaluate *op) {
     value = NULL;
 }
 
+void CodeGen_LLVM::visit(const Return *op) {
+    llvm::Value *value = codegen(op->value);
```

Comment:
Shouldn't this follow a similar path to an assertion failure and clean up allocations (or assert that there are none)?

More generally, is adding a new IR node really necessary? This seems like it fights with the convention that Halide-generated functions (and Halide runtime functions) return exit codes.